src/server/pps/server/api_server.go

```diff
@@ -34,6 +35,12 @@ import (
 	kube_labels "k8s.io/kubernetes/pkg/labels"
 )
 
+const (
+	// MaxPodsPerChunk is the maximum number of pods we can schedule for each
+	// chunk in case of failures.
+	MaxPodsPerChunk = 3
```

Comment:
This is basically the maximum number of retries for a chunk right?
If so it may make sense to have that be a configurable part of the Job/Pipeline Spec.
This isn't strictly necessary for merging this code though.